projects/step-frontend/src/lib/modules/timeseries/performance-view/filter-bar/filter-bar.component.ts

```diff
@@ -1,68 +1,137 @@
+  readonly FilterBarItemType = FilterBarItemType;
 
-  @Input() defaultFilterOptions: TsFilterItem[] = [];
+  filters: TsFilterItem[] = [];
+  moreFilters: TsFilterItem[] = [];
+
+  constructor(private _changeDetectorRef: ChangeDetectorRef) {}
 
   ngOnInit(): void {
     if (!this.context) {
       throw new Error('Context input is mandatory');
     }
+
     this.emitFilterChange$.pipe(debounceTime(this.EMIT_DEBOUNCE_TIME)).subscribe(() => {
-      this.onFiltersChange.emit(this.items.filter(FilterUtils.filterItemIsValid)); // we ignore empty filters
+      // we ignore empty filters
+      this.onFiltersChange.emit(this.filters.filter(FilterUtils.filterItemIsValid));
     });
   }
 
-  handleFilterChange(item: TsFilterItem) {
+  ngOnDestroy(): void {
+    this.emitFilterChange$.complete();
+    this.onFiltersChange.complete();
+  }
+
+  handleFilterChange() {
     this.emitFilterChange$.next();
   }
 
   addFilterItem(item: TsFilterItem) {
-    let filterExists = this.items.find((i) => i.attributeName === item.attributeName);
-    if (filterExists) {
-      return;
+    const filterIndex = this.filters.findIndex((i) => i.attributeName === item.attributeName);
+
+    if (filterIndex !== -1) {
+      const index =
+        filterIndex < this.filters.length - this.moreFilters.length
+          ? filterIndex
+          : filterIndex + this.moreFilters.length;
+
+      this.filterComponents!.toArray()[index].menuTrigger!.openMenu();
     } else {
-      this.items.push(JSON.parse(JSON.stringify(item))); // make a clone
+      this.addFilter(JSON.parse(JSON.stringify(item)));
     }
   }
 
   addCustomFilter(type: FilterBarItemType) {
-    this.items.push({
+    this.addFilter({
       attributeName: '',
       type: type,
       label: '',
     });
   }
 
   removeFilterItem(index: number) {
-    let itemToDelete = this.items[index];
-    this.items.splice(index, 1);
+    let itemToDelete = this.filters[index];
+
+    this.filters.splice(index, 1);
+
     if (FilterUtils.filterItemIsValid(itemToDelete)) {
       this.emitFilterChange$.next();
     }
   }
 
-  get FilterBarItemType() {
-    return FilterBarItemType;
+  initMoreFilters(): void {
+    if (!this.appliedFilters) {
+      return;
+    }
+
+    const orderedTops = this.appliedFilters
+      .map((appliedFilter) => {
+        const { top } = appliedFilter.nativeElement.getBoundingClientRect();
+
+        return top;
+      })
+      .sort((a, b) => a - b);
+
+    const sameLevelFilters = this.appliedFilters.filter((appliedFilter) => {
+      const { top } = appliedFilter.nativeElement.getBoundingClientRect();
+
+      return top === orderedTops[0];
+    });
+
+    this.moreFilters = this.filters.slice(sameLevelFilters.length);
   }
 
-  ngOnDestroy(): void {
-    this.emitFilterChange$.complete();
-    this.onFiltersChange.complete();
+  private addFilter(item: TsFilterItem): void {
+    this.filters.push(item);
+    this._changeDetectorRef.detectChanges();
+    this.initMoreFilters();
```

Comment:
1. a filter is added
2. we invoke ```detectChanges()``` in order for the UI to get updated
3. we invoke ```initMoreFilters()``` which is based on the previous change in the UI
4. having initialized ```moreFilters```, we invoke ```detectChanges()``` again
5. having both ```filters``` and ```moreFilters``` updated, we now invoke ```openMenu()``` for the last element of the ```filterComponents``` query list (the ```filterComponents``` query list holds both ```filters``` and ```moreFilters``` component references, so its a matter of a simple index calculation in order to show the menu for the proper filter, meaning if a filter is present in both ```div.applied-filters``` and ```div.more-applied-filters``` it would show the menu for the one in ```div.more-applied-filters```)